teps/0017-results-api.md

```diff
@@ -0,0 +1,711 @@
+---
+title: Tekton Results API
+authors:
+  - "@ImJasonH"
```

Comment:
FWIW you can feel free to mark me as a reviewer and yourself as the sole author.